tests/dispatch-claude-bg.test.sh

```
@@ -0,0 +1,136 @@
+  "version-gate error includes actionable npm install pointer"
+assert_grep "$SOLVE_PIPELINE" \
+  '^BG_PROMPT_MODE=argv' \
+  "Phase A hardcodes BG_PROMPT_MODE=argv (probe deferred per fix(solve) commit 0c17169 — claude --bg --help is not introspective in v2.1.139)"
+assert_grep "$SOLVE_PIPELINE" \
+  'TERMINAL_FLAG_DEPRECATED_NOTE' \
+  "Constants table defines TERMINAL_FLAG_DEPRECATED_NOTE"
+assert_grep "$SOLVE_PIPELINE" \
+  '^TERMINAL_FLAG_DEPRECATED_NOTE=' \
+  "Phase A binds TERMINAL_FLAG_DEPRECATED_NOTE as a bash variable (B1 regression guard)"
+assert_grep "$SOLVE_PIPELINE" \
+  'echo "\$TERMINAL_FLAG_DEPRECATED_NOTE" >&2' \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)